j2cl-tasks/src/main/java/com/vertispan/j2cl/tools/Javac.java

```diff
@@ -33,28 +40,70 @@
     StandardJavaFileManager fileManager;
     private DiagnosticCollector<JavaFileObject> listener;
 
-    public Javac(BuildLog log, File generatedClassesPath, List<File> sourcePaths, List<File> classpath, File classesDirFile, File bootstrap) throws IOException {
+    public Javac(BuildLog log, File generatedClassesPath, List<File> sourcePaths, List<File> classpath, File classesDirFile, File bootstrap, Mode mode) throws IOException {
         this.log = log;
-//        for (File file : classpath) {
-//            System.out.println(file.getAbsolutePath() + " " + file.exists() + " " + file.isDirectory());
-//        }
-        javacOptions = new ArrayList<>(Arrays.asList("-encoding", "utf8", "-implicit:none", "-bootclasspath", bootstrap.toString()));
+        compiler = ToolProvider.getSystemJavaCompiler();
+        listener = new DiagnosticCollector<>();
+        fileManager = compiler.getStandardFileManager(listener, null, StandardCharsets.UTF_8);
+        javacOptions = new ArrayList<>(Arrays.asList("-encoding", "utf8", "-implicit:none"));
+        if (log.isDebugEnabled()) {
+            javacOptions.add("-verbose");
```

Comment:
Hmm, I don't care for this - it means that the cached logs will stay non-verbose, even if you do a verbose log execution.

Instead it would be nicer to just not write the logs out if we can, but always ask for them?

Reply:
Right, so I leave the "-verbose" here always on? It helped a lot in debugging. On the other hand, the build certainly runs faster without it. Ideally, we would run mvn with "-X" and have a detailed log and run without "-X" and have a fast build.
Hmm. Leaving it in, for now.